refactor: migrate to @metamask/bridge-controller

micaelae opened the pull request:
<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been completely filled out, and PR status checks have passed at least once.
-->

## **Description**

This PR removes the BridgeController class and shared bridge utils from the extension repo, and replaces them with [@metamask/bridge-controller](https://www.npmjs.com/package/@metamask/bridge-controller) imports

See [Copilot summary for more details](https://github.com/MetaMask/metamask-extension/pull/31059#pullrequestreview-2712027334)

<!--
Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/31059?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Bridge and swap functionality for EVM and Solana should be unchanged, just updating imports
2. Compare behavior to latest released prod/builds

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

micaelae committed:
feat: migrate to @metamask/bridge-controller

micaelae committed:
refactor: replace solana conditions with isSolanaChainId

micaelae committed:
refactor: rm BridgeController

micaelae committed:
refactor: import bridge types, constants and utils from core

micaelae committed:
refactor: remove bridgeState from controller state

micaelae committed:
fix: type errors

micaelae committed:
refactor: use SWAPS_CHAINID_DEFAULT_TOKEN_MAP from core

micaelae committed:
fix: use bridge token endpoint for solana

micaelae committed:
fix: token alerts payload

micaelae committed:
chore: use token types from bridge-controller

micaelae commented on ui/hooks/bridge/useTokensWithFiltering.ts:
```diff
@@ -75,21 +83,34 @@ export const useTokensWithFiltering = (
-      // Use cached token data if updated in the last 10 minutes
-      if (timestamp && Date.now() - timestamp <= 10 * MINUTE) {
-        return cachedTokens[hexChainId]?.data;
+    if (chainId) {
```
This removes special handling for Solana since the bridge-api's getToken endpoint supports Solana tokens now

micaelae committed:
chore: clean up comments

micaelae committed:
refactor: use getNativeAssetForChainId

micaelae committed:
fix: undo using bridge-controller constants in useSwapDefaultToToken

micaelae committed:
fix: invalid type imports

micaelae committed:
refactor: caip formatter updates

micaelae committed:
chore: package updates

micaelae committed:
chore: use preview bridge-controller build

micaelae committed:
fix: e2e constants for bridge feature flags

micaelae committed:
fix: getDefaultBridgeControllerState

micaelae committed:
fix: circular dependency

micaelae committed:
fix: lavamoat policy

micaelae committed:
fix: unit tests

micaelae committed:
Merge branch 'main' into mms2102-bridge-migration

micaelae committed:
fix: missing import

micaelae committed:
fix: bridge-status mocks

micaelae committed:
fix: bridge selectors unit tests

micaelae committed:
fix: getFromToken selector

micaelae committed:
chore: bump bridge-controller

micaelae committed:
fix: unit tests

micaelae committed:
fix: lavamoat

micaelae committed:
fix: MultichainBridgeQuoteCard storybook

micaelae committed:
fix: useLatestBalance

micaelae committed:
chore: undo calcLatestSrcBalance change

micaelae committed:
chore: bump bridge-controller

micaelae committed:
Merge branch 'main' into mms2102-bridge-migration

micaelae committed:
Merge branch 'main' into mms2102-bridge-migration

micaelae committed:
fix: bridge-cta-button warnings

micaelae committed:
chore: memoize native assets

micaelae committed:
fix: warnings in useTokensWithFiltering

micaelae committed:
refactor: rm chainId from useLatestBalance

micaelae committed:
chore: use @metamask/bridge-controller 11.0.0

micaelae commented:
@metamaskbot  update-policies

micaelae committed:
fix: useTokensWithFiltering tests

micaelae committed:
Merge branch 'main' into mms2102-bridge-migration

micaelae commented:
@metamaskbot  update-policies

micaelae commented:
@metamaskbot update-policies

micaelae committed:
chore: update lavamoat

micaelae committed:
Merge branch 'main' into mms2102-bridge-migration

micaelae committed:
Merge branch 'main' into mms2102-bridge-migration

micaelae committed:
fix: sentry tests

micaelae committed:
fix: lavamoat

micaelae committed:
Merge branch 'main' into mms2102-bridge-migration

micaelae committed:
fix: e2e tests

micaelae committed:
chore: mock bridge calls

micaelae committed:
fix: add bridge featureFlags to allowed e2e urls

micaelae committed:
fix: sentry tests

micaelae commented on test/e2e/mock-e2e-allowlist.js:
```diff
@@ -29,6 +29,7 @@ const ALLOWLISTED_URLS = [
   'https://authentication.api.cx.metamask.io/siwe/verify',
   'https://bafkreifvhjdf6ve4jfv6qytqtux5nd4nwnelioeiqx5x2ez5yrgrzk7ypi.ipfs.dweb.link/',
   'https://bafybeidxfmwycgzcp4v2togflpqh2gnibuexjy4m4qqwxp7nh3jx5zlh4y.ipfs.dweb.link/1.json',
+  'https://bridge.api.cx.metamask.io/getAllFeatureFlags',
```
@MetaMask/extension-platform team can I get a review for this?

dbrans replied:
Hi @micaelae – from the comment at the top of this file, it looks like we're actually trying to phase it out in favor of mocked e2e tests:

```js
// Please do not add any more items to this list.
// This list is temporary and the goal is to reduce it to 0, meaning all requests are mocked in our e2e tests.
```

Do you want to add a mockttp request for this url?

leotm reviewed (dismissed):
@metamask/bridge-controller@11.0.0 added and 5 mm resources to policy
nb: global fetch added in @metamask/bridge-controller>@metamask/snaps-utils
(bridge-controller only imports HandlerType from @metamask/snaps-utils)

micaelae committed:
fix: propagate abort signal to fetch function

micaelae committed:
Merge branch 'main' into mms2102-bridge-migration

micaelae committed:
chore: add useMemo dep

micaelae committed:
Merge branch 'main' into mms2102-bridge-migration

micaelae committed:
Merge branch 'main' into mms2102-bridge-migration

micaelae committed:
Merge branch 'main' into mms2102-bridge-migration

naugtur reviewed (dismissed):
Approving on behalf of policy-reviewers. (Since I'm already looking and it's obviously fine, but next time let's follow the process)

All additions to policy are metamask packages. Nothing suspicious.

naugtur commented:
The bug in `MetaMask/action-checkout-and-setup@v1` action that would pull node_modules from cache for the main branch instead of the current checkout caused policy update to fail here: https://github.com/MetaMask/metamask-extension/actions/runs/14042981658/job/39317544345 
Jyoti reported a similar issue yesterday and we found the bug.

I see you've updated the policy manually and pushed. :+1: 

I'll rerun the policy generation (expecting no changes) to see that it now works correctly.

micaelae committed:
Merge branch 'main' into mms2102-bridge-migration

dbrans commented:
Nice work, @micaelae. I left a little feedback [here](https://github.com/MetaMask/metamask-extension/pull/31059/files#r2017171137).

micaelae committed:
Merge branch 'main' into mms2102-bridge-migration

micaelae committed:
Merge branch 'main' into mms2102-bridge-migration

micaelae committed:
chore: add network names for caip chainids (#31372)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**
Some components use hex and some use caip so adding network name
mappings for both chainId formats

<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

[![Open in GitHub
Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/31372?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

micaelae committed:
Merge branch 'main' into mms2102-bridge-migration

dbrans reviewed (approved):
Approving since this needs to make the release cut.
